add more docs about scatter/gather (#820)
nf-core/sarek: docs/usage.md

In addition, currently the mismatch penalty for reads with tumor status in the sample sheet are mapped with a mismatch penalty of `-B 3`.

## How to manage scatter/gathering (parallelization with-in each sample)

> While nextflow ensures all samples are run in parallel, the pipeline in addition attempts to split input files for one sample into smaller chunks and processes these in parallel.

Reviewer: ```suggestion
While Nextflow ensures all samples are run in parallel, the pipeline can split input files for each sample into smaller chunks which are processes in parallel.
```

> This allows on the one hand to speed up analysis for individual chunks, but might occupy more storage space.

Reviewer: ```suggestion
This speeds up analysis for individual chunks, but might occupy more storage space.
```

Therefore, the different scatter/gather options can be set by the user:

### Split Fastq files

By default, the input fastq files are split into smaller chunks with FASTP, mapped in parallel, and then merged and duplicate marked. This can be customized by setting the parameter `--split_fastq`.
This parameter determines how many reads are within each split. Setting it to `0` will turn of any splitting and only one mapping process is run per input fastq file.

> > FastP always creates as many chunks as CPUs are specified (by default 12). Thus, the parameter `--split_fastq` is an upper bound, e.g. if 1/12th of the Fastq file exceeds the provided value another fastq file will be generated.

Reviewer: ```suggestion
> **NB:** FASTP can only create as many chunks as CPUs are specified (by default 12). Thus, the parameter `--split_fastq` has an upper bound, e.g. if 1/12th of the fastq file exceeds the provided value another fastq file will be generated.
```

### Intervals for Base Quality Score Recalibration and Variantcalling

> The pipeline attempts to parallelize base quality score recalibration and variant calling where possible across genomic chunks of roughly similar sizes of one samples. For this, a bed file with interesting genomic regions is used. By default, the intervals file for WGS provided by GATK is used (details [here](https://gatk.broadinstitute.org/hc/en-us/articles/360035889551-When-should-I-restrict-my-analysis-to-specific-intervals-)). When running targeted analysis, it is usually good to use the bed file containing the targeted regions.

Reviewer: ```suggestion
The pipeline can parallelize base quality score recalibration and variant calling across genomic chunks of roughly similar sizes.
For this, a bed file containing genomic regions of interest is used, it's the intervals file.
By default, the intervals file for WGS used is the one provided by GATK (details [here](https://gatk.broadinstitute.org/hc/en-us/articles/360035889551-When-should-I-restrict-my-analysis-to-specific-intervals-)).
When running targeted analysis, it is recommended to use the bed file containing the targeted regions.
```

The amount of scatter/gathering can be customized by adjusting the parameter `--nucleotides_per_second`.

> > **NB:** The _same_ intervals are processed regardless of the number of groups. The number of groups however determines over how many compute nodes the analysis is scattered.

Reviewer: ```suggestion
> **NB:** The _same_ intervals are processed regardless of the number of groups. The number of groups however determines over how many compute nodes the analysis is scattered on.
```
